acs-engine.go

```diff
@@ -147,6 +155,22 @@ func main() {
 		os.Exit(1)
 	}
 
+	if (len(*caCertificatePath) > 0 && len(*caKeyPath) == 0) ||
```

Comment:
Doing this validation here means that it only applies when they are user-supplied flags.

If the user puts in the CA private key, but not the CA certificate, then they will not hit this logic, and instead will wind up in a situation where their private key is ignored because of your logic below: `len(a.CertificateProfile.CaCertificate) != 0 && len(a.CertificateProfile.GetCAPrivateKey()) != 0`.